doc/compiler/regular_expression/Regular_Expression.md

```diff
@@ -0,0 +1,41 @@
+# Regular expression
```

Comment:
In general, I think it would be better for review comments if newlines were used to split these paragraphs across multiple lines.  This allows review comments to be more closely associated with the text they apply to, as opposed to the entire paragraph.